camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -340,9 +360,9 @@ public DslConfigSupplier(DslComponent component, String keyName) {
         public final Maybe<Object> getImmediately() {
             Maybe<Entity> targetEntityMaybe = component.getImmediately();
             if (targetEntityMaybe.isAbsent()) return Maybe.absent("Target entity not available");
-            Entity targetEntity = targetEntityMaybe.get();
+            EntityInternal targetEntity = (EntityInternal) targetEntityMaybe.get();
             
-            return Maybe.of(targetEntity.getConfig(ConfigKeys.newConfigKey(Object.class, keyName)));
+            return targetEntity.config().getNonBlocking(ConfigKeys.newConfigKey(Object.class, keyName));
```

Comment:
Nice, forgot we have this method. Now take a guess at how it's implemented :)
The [simple](https://github.com/apache/brooklyn-server/blob/master/core/src/main/java/org/apache/brooklyn/core/objs/AbstractConfigurationSupportInternal.java#L133) case looks easy enough to update. The [structure key](https://github.com/apache/brooklyn-server/blob/master/core/src/main/java/org/apache/brooklyn/core/objs/AbstractConfigurationSupportInternal.java#L91) case looks like a rabbit hole though (with an all too familiar comment).
I'm fine with leaving the structured key case as is for this PR, may be log a warning.